pyroSAR/drivers.py

```diff
@@ -1669,102 +2372,152 @@ def insert(self, scene_in, verbose=False, test=False):
+            session.rollback()
         print('{} scenes registered regularly'.format(counter_regulars))
-        print('{} duplicates detected and registered'.format(counter_duplicates))
-    
+        print('{} duplicates registered'.format(counter_duplicates))
+        if len(list_duplicates) != 0:
+            print('Scene(s) {} already registered as duplicate!'.format(list_duplicates))
+
     def is_registered(self, scene):
         """
         Simple check if a scene is already registered in the database.
 
         Parameters
         ----------
-        scene: str
+        scene: str or ID
             the SAR scene
 
         Returns
         -------
         bool
             is the scene already registered?
         """
-        return len(self.select(scene=scene)) != 0 or len(self.select_duplicates(scene=scene)) != 0
-    
+        id = scene if isinstance(scene, ID) else identify(scene)
+        exists_data = self.Session().query(self.Data.scene).filter(self.Data.scene == id.scene).first()
+        exists_duplicates = self.Session().query(self.Duplicates.scene).filter(
+            self.Duplicates.scene == id.scene).first()
+        in_data = False
+        in_dup = False
+        if exists_data:
+            in_data = len(exists_data) != 0
+        if exists_duplicates:
+            in_dup = len(exists_duplicates) != 0
+        return in_data or in_dup
+
+    def __is_registered_in_duplicates(self, scene):
+        """
+        Simple check if a scene is already registered in the database.
+
+        Parameters
+        ----------
+        scene: str or ID
+            the SAR scene
+
+        Returns
+        -------
+        bool
+            is the scene already registered?
+        """
+        id = scene if isinstance(scene, ID) else identify(scene)
+        exists_duplicates = self.Session().query(self.Duplicates.scene).filter(
+            self.Duplicates.scene == id.scene).first()
+        in_dup = False
+        if exists_duplicates:
+            in_dup = len(exists_duplicates) != 0
+        return in_dup
+
     def cleanup(self):
         """
         Remove all scenes from the database, which are no longer stored in their registered location
-        
+
         Returns
         -------
 
         """
-        cursor = self.conn.cursor()
         for table in ['data', 'duplicates']:
             missing = self.__select_missing(table)
             for scene in missing:
-                query = '''DELETE FROM {0} WHERE scene=?'''.format(table)
-                cursor.execute(query, (scene,))
-        self.conn.commit()
-    
+                print(scene)
+                print(table)
+                self.drop_element(scene, table)
+
     @staticmethod
     def encode(string, encoding='utf-8'):
         if not isinstance(string, str):
             return string.encode(encoding)
         else:
             return string
-    
-    def export2shp(self, shp):
+
+    def export2shp(self, path):
```

Comment:
Better request a name for a shapefile with parameter `shp` and add a new parameter `table` with default 'data'.  
This way a user has more control over which files are produced and how they are named.  
A user might not be interested in the duplicates table anyways.

Reply:
duplicates does not have spatial information anyway, changed to only export data table.

Reply:
any update on this?

Reply:
resolved with subprocess, now per default data is exported, by choosing duplicates it will export a dbf file.